src/Microsoft.AspNet.Localization/RequestLocalizationMiddleware.cs

```diff
@@ -142,6 +156,39 @@ private CultureInfo GetCultureInfo(IList<string> cultures, IList<CultureInfo> su
                 }
             }
 
+            if (fallbackToAncestorCulture & currentDepth < MaxCultureFallbackDepth)
+            {
+                // Walk backwards through the culture list and remove any root cultures (those with no parent)
+                for (var i = cultureNames.Count - 1; i >= 0; i--)
+                {
+                    var cultureName = cultureNames[i];
+                    if (cultureName != null)
+                    {
+                        var lastIndexOfHyphen = cultureName.LastIndexOf('-');
+                        if (lastIndexOfHyphen > 0)
```

Comment:
The point is to avoid creating instances of `CultureInfo` at all because it's expensive.

We're not overly concerned on what is strictly valid in this case, as ultimately it's checked against the list of cultures the application explicitly stated it supported.